staker-rewards/lib/rewards-calculator.js

```diff
@@ -57,20 +62,24 @@ export default class RewardsCalculator {
     const operatorsRewards = []
 
     for (const operatorRewardsFactors of operatorsRewardsFactors) {
-      const { rewardWeight } = operatorRewardsFactors
+      const { rewardWeight, requirementsViolations } = operatorRewardsFactors
       const rewardRatio = rewardWeightSum.isGreaterThan(new BigNumber(0))
         ? rewardWeight.dividedBy(rewardWeightSum)
         : new BigNumber(0)
 
-      const totalRewards = this.interval.totalRewards.multipliedBy(rewardRatio)
+      const totalRewards =
+        requirementsViolations.length > 0
+          ? new BigNumber(0)
+          : this.interval.totalRewards.multipliedBy(rewardRatio)
```

Comment:
Just to make sure I understand the mechanism correctly: if one operator violated rules, what's happening with their share of rewards? Is it distributed among other operators?

Reply:
Nope, it remains in the pool.

Reply:
How hard would be to change it so that it's distributed between other operators?

Reply:
Quite easy I think. We must set operator weight to `0` so it doesn't count in the rewards weight sum.

Reply:
OK, let's leave it as-is and we'll discuss it separately.

Reply:
(I also think it's a better approach than distributing this reward between operators from the current interval as this way we do not incentivize misbehavior DoSing other operators to increase one's reward.)